src/plugins/persistence/couch/CouchObjectProvider.js

```diff
@@ -334,103 +390,55 @@ export default class CouchObjectProvider {
+            //TODO: handle aborting fetch requests
+            // const signal = controller.signal;
 
-        let reader;
+            if (typeof this.stopObservingObjectChanges === 'function') {
+                this.stopObservingObjectChanges();
+            }
 
-        if (response.body === undefined) {
-            error = true;
-        } else {
-            reader = response.body.getReader();
-        }
+            this.stopObservingObjectChanges = () => {
+                controller.abort();
+                delete this.stopObservingObjectChanges;
+            };
 
-        while (!error) {
-            const {done, value} = await reader.read();
-            //done is true when we lose connection with the provider
-            if (done) {
-                error = true;
-            }
+            // feed=continuous maintains an indefinitely open connection with a keep-alive of HEARTBEAT milliseconds until this client closes the connection
+            // style=main_only returns only the current winning revision of the document
+            let url = `${this.url}/_changes?feed=continuous&style=main_only&heartbeat=${HEARTBEAT}`;
 
-            if (value) {
-                let chunk = new Uint8Array(value.length);
-                chunk.set(value, 0);
-                const decodedChunk = new TextDecoder("utf-8").decode(chunk).split('\n');
-                if (decodedChunk.length && decodedChunk[decodedChunk.length - 1] === '') {
-                    decodedChunk.forEach((doc, index) => {
-                        try {
-                            const object = JSON.parse(doc);
-                            object.identifier = {
-                                namespace: this.namespace,
-                                key: object.id
-                            };
-                            let keyString = this.openmct.objects.makeKeyString(object.identifier);
-                            let observersForObject = this.observers[keyString];
-
-                            if (observersForObject) {
-                                observersForObject.forEach(async (observer) => {
-                                    const updatedObject = await this.get(object.identifier);
-                                    observer(updatedObject);
-                                });
-                            }
-                        } catch (e) {
-                            //do nothing;
-                        }
-                    });
-                }
+            let body = {};
+            if (filter) {
+                url = `${url}&filter=_selector`;
+                body = JSON.stringify(filter);
             }
 
-        }
-
-        if (error && Object.keys(this.observers).length > 0) {
-            this.observeObjectChanges();
+            //TODO: How do we abort the request here? SharedWorkers don't allow cloning signals
+            this.changesFeedSharedWorker.port.postMessage({
```

Comment:
The abort would not be done from here because you don't have all of the information to abort the connection from here. The worker would abort the fetch when there are no more listeners. This is an optimization to avoid holding connections open when there are no listeners.

Reply:
Done.